### PR TITLE
Volcano plot: update 

### PR DESCRIPTION
**Pull request by mblue9**
FOR CONTRIBUTOR:
* [ ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by bgruening** [comment]
Thanks @mblue9. Is this tool still WIP? Feel free to merge if not.

**Review by bgruening — APPROVED** [review]

File: tools/volcanoplot/volcanoplot.xml [hunk 1]
```diff
@@ -116,13 +248,29 @@ Rscript '${__tool_directory__}/volcanoplot.R'
...
+            <param name="lfc_col" value="2" />
+            <param name="label_col" value="1" />
+            <param name="lfc_thresh" value="0" />
+            <param name="label_select" value="file"/>
+            <param name="label_file" ftype="tabular" value="labels.tab" />
+            <param name="rscript_out" value="True"/>
+            <param name="rdata_out" value="True"/>
+            <output name="plot" value= "out2.pdf" compare="sim_size" />
+            <output name="rscript" value= "out.rscript" lines_diff="8"/>
+            <output name="rdata" value= "out.rdata" compare="sim_size" />
```
Comment (bgruening) [inline tools/volcanoplot/volcanoplot.xml:265]: `assert_content` has now a paramter called `has_size` https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-tests-test-output-assert-contents that could be used here. This way we don't need to store out.rdata on github.

In theory we can also use that for PDF

**Comment by mblue9** [comment]
Thanks! Out of WIP now. Made some changes to allow the R code to be output as there were a few requests for that in the tutorial feedback. Was thinking could be good to show how to edit the R code in Galaxy RStudio (e.g. change the colours) either in the current volcano plot tutorial or as a separate mini-tutorial.

Will I merge or should I let someone have a look at the new changes?

File: tools/volcanoplot/volcanoplot.xml (hunk 1 above)
Comment (mblue9) [reply tools/volcanoplot/volcanoplot.xml:265]: Ok great, I'll take a look at that.

File: tools/volcanoplot/volcanoplot.xml [hunk 2]
```diff
@@ -116,13 +248,29 @@ Rscript '${__tool_directory__}/volcanoplot.R'
             <param name="label_file" ftype="tabular" value="labels.tab" />
             <output name="plot" value= "out2.pdf" compare="sim_size" />
         </test>
+        <test>
```
Comment (bgruening) [inline tools/volcanoplot/volcanoplot.xml:251]: can you add to your tests `expected_num_outputs` please: https://docs.galaxyproject.org/en/latest/dev/schema.html#id86

This tests the output-filters :)

**Comment by bgruening** [comment]
Two last small comments :)

Thanks Maria!

**Comment by mblue9** [comment]
Changes made, thanks for the review!

**Review by bgruening — COMMENTED** [review]
Looks good to me! Can you remove the unneeded files please. out.rdata can now be removed, left?

When you merge please note that you can select -squash-and-merge, this will get rid of the data file also in your commit history. Otherwise, even if you remove it the file will stay in the git history and will blow up the size of this repo.

Go raibh maith agat!

**Comment by mblue9** [comment]
>Looks good to me! Can you remove the unneeded files please. out.rdata can now be removed, left?

Oops thought I had.

>When you merge please note that you can select -squash-and-merge, this will get rid of the data file also in your commit history. Otherwise, even if you remove it the file will stay in the git history and will blow up the size of this repo.

Will do!

>Go raibh maith agat!

😎

**Pull request merged**